### PR TITLE
Update Jest to v20

### PR DESCRIPTION
**Pull request by gabrielecirulli**
Updates to Jest 20 by:

 - Updating snapshots where they subtly changed due to changes in the tool
 - Moving the config to jest.config.js

One of the primary reasons I wanted to do this before working on it further is that under certain versions of Jest and Node, `console.log` calls will be muted and this makes testing quite hard. Let me know if you'd like to make any further changes. I'll probably base another branch on this so that I get a better testing flow, and potentially rebase onto the old version if needed.

### CODE REVIEW

**Comment by gabrielecirulli** [comment]
It looks like the tests fail. I used Node 8 instead of 6 which travis does. Have you considered hardcoding the version (e.g. in a .node-version file) and/or updating it to v8? I can help do it, should I do it directly in this PR?

**Comment by gabrielecirulli** [comment]
I took the liberty of updating the node version in this PR just to see if it resolves the test issues. Depending on what you prefer, this can still be split into multiple PRs or merged as one.

**Comment by gabrielecirulli** [comment]
Now getting a strange `npm ERR! ELOOP: too many symbolic links encountered, access 'node_modules/npm/node_modules/has-unicode'`.

I don't really know what would cause that. Any clues?

**Comment by gabrielecirulli** [comment]
@jongold any update on whether this is interesting to you? If so I can pursue it further and maybe try my hand at the symbol layout issue in #114

File: .node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+8.1.0
```
Comment (ljharb) [inline .node-version]: This file is only used by nodenv; travis-ci uses nvm. Also is v8.1 even out?

Please remove this file.

File: .travis.yml [hunk 2]
```diff
@@ -5,8 +5,7 @@ cache:
 notifications:
   email: false
 node_js:
-  - '7'
-  - '6'
```
Comment (ljharb) [inline .travis.yml]: Please also restore v6; dropping tests is dropping support, and we should be testing on v6 at least until it's not LTS.

File: __tests__/components/__snapshots__/RedBox.js.snap [hunk 3]
```diff
@@ -73,8 +69,9 @@ exports[`<RedBox /> renders simple errors 1`] = `
           "fontFamily": "Monaco",
           "marginTop": 20,
         }
-      }>
-      attemptAsync
+      }
+    >
```
Comment (ljharb) [inline __tests__/components/__snapshots__/RedBox.js.snap]: All these style changes should go in a separate PR.

File: package.json [hunk 4]
```diff
@@ -5,6 +5,10 @@
   "main": "lib/index.js",
   "license": "MIT",
   "author": "Jon Gold <jon.gold@airbnb.com> (http://jon.gold)",
+  "engines": {
```
Comment (ljharb) [inline package.json]: engines is advisory only, so it doesn't add much value; that said, this should work on node 6 and npm 3.

File: __tests__/components/__snapshots__/RedBox.js.snap (hunk 3 above)
Comment (ljharb) [inline __tests__/components/__snapshots__/RedBox.js.snap]: Ah sorry, ignore this, I didn't realize they were snapshots.

File: .node-version (hunk 1 above)
Comment (gabrielecirulli) [inline .node-version]: The current version of Node seems to be 8.1.1, yeah. I'll remove this.

**Comment by gabrielecirulli** [comment]
@ljharb what should we do about package-lock.json?

**Comment by gabrielecirulli** [comment]
Strangely, the build on Node 6 fails with some bad matches in the snapshots, probably due to changes in internal Node call stacks. On Node 8 it fails with `npm ERR! ELOOP: too many symbolic links encountered, access 'node_modules/npm/node_modules/has-unicode'.`, which as I mentioned couldn't find anything on how to solve. Any ideas?

**Comment by ljharb** [comment]
We should add `package-lock=false` to `.npmrc` and gitignore lockfiles, since only apps should have lockfiles. I'll do that in a separate commit now.

**Comment by gabrielecirulli** [comment]
Apologies, I'm abandoning this as I won't be able to dedicate any further time to it.
